nshlib/nsh_parse.c

```diff
@@ -603,24 +603,31 @@ static int nsh_execute(FAR struct nsh_vtbl_s *vtbl,
     {
       FAR char *sh_argv[4];
       FAR char *sh_cmd = "sh";
-      int i;
+      char cmd_str[CONFIG_NSH_LINELEN];
 
       DEBUGASSERT(strncmp(argv[0], sh_cmd, 3) != 0);
 
-      sh_argv[0] = sh_cmd;
-      sh_argv[1] = "-c";
-      for (i = 0; i < argc - 1; i++)
+      memset(cmd_str, 0, sizeof(cmd_str));
+
+      for (ret = 0; ret < argc; ret++)
         {
-          FAR char *p_arg = argv[i];
-          size_t len = strlen(p_arg);
+          if (sizeof(cmd_str) - strlen(cmd_str) - 1 <
```

Comment:
let's skip the check since it's impossible to larger than CONFIG_NSH_LINELEN

Reply:
get, done